django_setup_configuration/config_settings.py

```diff
@@ -0,0 +1,227 @@
+class ConfigField:
+    name: str
+    verbose_name: str
+    description: str
+    default_value: str
+    field_description: str
+
+
+class ConfigSettings:
+    """
+    Settings for configuration steps, used to generate documentation.
+
+    Attributes:
+        namespace (`str`): the namespace of configuration variables for a given
+            configuration
+        file_name (`str`): the name of the file where the documentation is stored
+        models (`list`): a list of models from which documentation is retrieved
+        update_field_descriptions (`bool`): if `True`, custom model fields
+            (along with their descriptions) are loaded via the settings variable
+            `DJANGO_SETUP_CONFIG_CUSTOM_FIELDS`
+        required_settings (`list`): required settings for a configuration step
+        optional_settings (`list`): optional settings for a configuration step
+        detailed_info (`dict`): information for configuration settings which are
+            not associated with a particular model field
+
+    Example:
+        Given a configuration step `FooConfigurationStep`: ::
+
+        FooConfigurationStep(BaseConfigurationStep):
+            verbose_name = "Configuration step for Foo"
+            enable_setting = "FOO_CONFIG_ENABLE"
+            config_settings = ConfigSettings(
+                namespace="FOO",
+                file_name="foo",
+                models=["FooConfigurationModel"],
+                required_settings=[
+                    "FOO_SOME_SETTING",
+                    "FOO_SOME_OTHER_SETTING",
+                ],
+                optional_settings=[
+                    "FOO_SOME_OPT_SETTING",
+                    "FOO_SOME_OTHER_OPT_SETTING",
+                ],
+                detailed_info={
+                    "example_non_model_field": {
+                        "variable": "FOO_EXAMPLE_NON_MODEL_FIELD",
+                        "description": "Documentation for a field that could not
+                            be retrievend from a model",
+                        "possible_values": "string (URL)",
+                    },
+                },
+            )
+    """
+
+    namespace: str
+    file_name: str
+    models: list[Type[models.Model]] | None
+    required_settings: list[str] = []
+    optional_settings: list[str] = []
+    detailed_info: dict[str, dict[str, str]] | None
+
+    def __init__(self, *args, update_field_descriptions: bool = False, **kwargs):
+        self.config_fields: list[ConfigField] = []
+
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        if not getattr(self, "models", None):
+            return
+
+        # add support for custom fields like PrivateMediaField
+        if update_field_descriptions:
+            self.update_field_descriptions()
+
+        for model in self.models:
+            self.create_config_fields(model=model)
+
+    @staticmethod
+    def get_default_value(field: models.Field) -> str:
```

Comment:
I'm not sure why, but my type checker complains about the usage of `models.Field` and `models.Model` in type hints, does this happen for you as well? I think it can be fixed by importing them directly `from django.db.models import Field, Model` and using them `Type[Field]`

![Screenshot from 2024-06-28 10-17-42](https://github.com/maykinmedia/django-setup-configuration/assets/29249171/2eca7585-410b-464c-9a1b-fed7113115df)

Reply:
Pycharm doesn't complain here